### PR TITLE
Refactor async tests under `test/unit/lib/plugins/aws`

### PR DESCRIPTION
**Pull request by sleepwithcoffee**
Address: #8368

### CODE REVIEW

File: test/unit/lib/plugins/aws/config-credentials.test.js [hunk 1]
```diff
@@ -110,26 +110,26 @@ describe('AwsConfigCredentials', () => {
...
       awsConfigCredentials.configureCredentials().then(() => {
         expect(awsConfigCredentials.options.profile).to.equal('default');
       }));
 
-    it('should resolve if the provider option is not "aws"', (done) => {
+    it('should resolve if the provider option is not "aws"', async () => {
       awsConfigCredentials.options.provider = 'invalid-provider';
 
-      awsConfigCredentials.configureCredentials().then(() => done());
+      expect(awsConfigCredentials.configureCredentials()).to.be.eventually.fulfilled;
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/config-credentials.test.js:129]: I think we need to `return` here

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @sleepwithcoffee. I've just spotted one issue

File: test/unit/lib/plugins/aws/config-credentials.test.js (hunk 1 above)
Comment (sleepwithcoffee) [reply test/unit/lib/plugins/aws/config-credentials.test.js:129]: sure, let me fix this.
while I was going through the tests with promise, I notice that they are written either use `await` or return promise. Maybe we can decide on one and consolidate from now on.

**Comment by sleepwithcoffee** [comment]
Updated, I believe this PR will wrap up all files under `test/unit/lib/plugins/aws`

File: test/unit/lib/plugins/aws/config-credentials.test.js (hunk 1 above)
Comment (medikoo) [reply test/unit/lib/plugins/aws/config-credentials.test.js]: Probably `await` is cleaner, but `return` is totally acceptable as well

**Review by medikoo — APPROVED** [review]
Thank you 👍 !

**Pull request merged**